src/cmd/run.js

```diff
@@ -107,33 +108,37 @@ export function defaultFirefoxClient(
     maxRetries=250, retryInterval=120,
   }: CreateFirefoxClientParams = {}
 ): Promise<RemoteFirefox> {
-  var retries = 0;
-
-  function establishConnection() {
-    return new Promise((resolve, reject) => {
-      connectToFirefox()
-        .then((connectedClient) => {
-          log.debug('Connected to the Firefox debugger');
-          resolve(connectedClient);
-        })
-        .catch(onlyErrorsWithCode('ECONNREFUSED', (error) => {
-          if (retries >= maxRetries) {
-            log.debug('Connect to Firefox debugger: too many retries');
-            throw error;
-          } else {
-            setTimeout(() => {
-              retries ++;
-              log.debug(
-                `Retrying Firefox (${retries}); connection error: ${error}`);
-              resolve(establishConnection());
-            }, retryInterval);
-          }
-        }))
-        .catch((error) => {
+  async function establishConnection() {
+    var lastError;
+
+    function* range(min: number, max: number, step: number = 1) {
+      for (let i = min; i <= max; i += step) {
+        yield i;
+      }
+    }
+
+    for (const retries of range(0, maxRetries)) {
```

Comment:
on a second look, I think this could just be the following (unless I'm missing something):

    for (let retries = 0; retries <= maxRetries; retries++) {
    }